packages/@aws-cdk/aws-s3-deployment/test/integ.bucket-deployment.ts

```diff
@@ -72,6 +72,13 @@ class TestBucketDeployment extends cdk.Stack {
       retainOnDelete: false,
     });
 
+    new s3deploy.BucketDeployment(this, 'DeployMeWithoutExtractingFilesOnDestination', {
+      sources: [s3deploy.Source.asset(path.join(__dirname, 'my-website'))],
+      destinationBucket,
+      extract: false,
+      retainOnDelete: false,
+    });
+
```

Comment:
This should probably have some sort of assertion to test that it's both uploaded properly and that it hasn't been extracted.

Reply:
I made use of the `AwsApiCall` construct and assertions feature provided by the integ-tests module to assert that the archive for the website folder is in the target bucket after deployment of the integ stack. Let me know if this is a solid assertion and proper use of the testing constructs.

Reply:
I think this looks reasonable.